stellargraph/core/graph.py

```diff
@@ -756,8 +758,19 @@ def node_features(self, nodes, node_type=None):
 
             node_type = types[0]
 
+        # handles eager execution and non-eager execution
+        if tf.executing_eagerly():
```

Comment:
Hm, this is pretty complicated... and I suspect the `else` case is catastrophically slow? (Something that can be verified with the `test_benchmark_get_features`.)

Do you think this is particularly valuable to support?

Reply:
yeah the `else` is catastrophically slow here :( 

I we dont support it then `subgraph` and `to_networkx` won't work in non-eager mode. fullbatch methods also wouldn't work as is but refactoring them to handle tensors natively would fix this

Reply:
i was planning on doing the fullbatch refactoring in a seperate PR but maybe its worth just doing it here, thoughts?

Reply:
Unless it's very complicated, I think it's fine to do here; if this PR was (at least, in our heads) reframed to be "switch StellarGraph to storing tensorflow tensors instead of numpy arrays".